client/components/feature-modules/blocks/components/render/reference/entity/reference-list.tsx

```
@@ -0,0 +1,105 @@
+
+    const references = hydrationResult?.references || [];
+
+    return (
+        <div className="space-y-3">
+            {items.map((item) => {
+                const reference = references.find((ref) => ref.entityId === item.id);
+                const onRemove = () => {
+                    handleRemoveEntity(item.id);
+                };
+                // Quick actions for this entity
+                const quickActions = useMemo(
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)